build/testdata/002.build.golden

```diff
@@ -1,5 +1,5 @@
 cc_test(
-    name = "b\"ar'\"",
+    name = 'b\"ar\'"',
```

Comment:
Why do we escape the first `"` and not the second?

Reply:
The original string is also `'b\"ar\'"'`. Buildifier currently doesn't remove unnecessary escapes: `"'\'"` is not changed during formatting.

Reply:
You mean, inside single-quoted strings?

Reply:
In this example `\"` should be actually replaced with unescaped `"`, at least that's how the Bazel interpreter parses strings.

Reply:
Oh right. The backslash can be dropped here.

The bug I was mentioning is for other characters: `"a\w" == "a\\w"`.